### PR TITLE
Do not prevent taking the switch /NodeReuse when FEATURE_NODE_REUSE i…

### PR DESCRIPTION
**Pull request by mfilippov**
…s OFF, but constrain the switch values to never be True. This way existing build scripts stay compatible with an MSBuild installation compiled without FEATURE_NODE_REUSE if they were passing /NodeReuse:False explicitly.

### CODE REVIEW

File: src/MSBuild/Resources/Strings.resx [hunk 1]
```diff
@@ -989,6 +989,14 @@ Copyright (C) Microsoft Corporation. All rights reserved.
      LOCALIZATION: The prefix "MSBUILD : error MSBxxxx:" should not be localized.
   </comment>
   </data>
+  <data name="InvalidNodeReuseTrueValue" UESanitized="true" Visibility="Public">
+    <value>MSBUILD : error MSB1042: Node reuse value is not valid. This installation has been built without the NodeReuse feature, and only takes False for the node reuse switch value.</value>
+    <comment>{StrBegin="MSBUILD : error MSB1042: "}
+     UE: This message does not need in-line parameters because the exception takes care of displaying the invalid arg.
+     This error is shown when a user specifies a node reuse value that is not equivilant to Boolean.TrueString or Boolean.FalseString.
```
Comment (rainersigwald) [inline src/MSBuild/Resources/Strings.resx]: typo: `equivilant` -> `equivalent`

File: src/MSBuild/Resources/Strings.resx [hunk 2]
```diff
@@ -989,6 +989,14 @@ Copyright (C) Microsoft Corporation. All rights reserved.
      LOCALIZATION: The prefix "MSBUILD : error MSBxxxx:" should not be localized.
   </comment>
   </data>
+  <data name="InvalidNodeReuseTrueValue" UESanitized="true" Visibility="Public">
+    <value>MSBUILD : error MSB1042: Node reuse value is not valid. This installation has been built without the NodeReuse feature, and only takes False for the node reuse switch value.</value>
```
Comment (rainersigwald) [inline src/MSBuild/Resources/Strings.resx]: This seems a bit too detailed. Maybe something like

```
MSBUILD : error MSB1042: Node reuse value is not valid. This version of MSBuild does not support node reuse. If specified, the node reuse switch value must be false.
```

?

File: src/MSBuild/CommandLineSwitches.cs [hunk 3]
```diff
@@ -94,9 +94,7 @@ internal enum ParameterizedSwitch
             FileLoggerParameters7,
             FileLoggerParameters8,
             FileLoggerParameters9,
-#if FEATURE_NODE_REUSE
-            NodeReuse,
-#endif
+            NodeReuse, // Should not hide this option when FEATURE_NODE_REUSE is turned off, otherwise existing build scripts which pass /NodeReuse:False would start failing even though they're fully compatible with a build without FEATURE_NODE_REUSE
```
Comment (rainersigwald) [inline src/MSBuild/CommandLineSwitches.cs]: I think you can drop this comment; it'll be in the commit description and in the handling code.

File: src/MSBuild/CommandLineSwitches.cs [hunk 4]
```diff
@@ -274,9 +272,8 @@ bool emptyParametersAllowed
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters7", "flp7" },     ParameterizedSwitch.FileLoggerParameters7,      null,                           false,          "MissingFileLoggerParameterError",     true,   false  ),
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters8", "flp8" },     ParameterizedSwitch.FileLoggerParameters8,      null,                           false,          "MissingFileLoggerParameterError",     true,   false  ),
             new ParameterizedSwitchInfo(  new string[] { "fileloggerparameters9", "flp9" },     ParameterizedSwitch.FileLoggerParameters9,      null,                           false,          "MissingFileLoggerParameterError",     true,   false  ),
-#if FEATURE_NODE_REUSE
+// Should not hide this option when FEATURE_NODE_REUSE is turned off, otherwise existing build scripts which pass /NodeReuse:False would start failing even though they're fully compatible with a build without FEATURE_NODE_REUSE
```
Comment (rainersigwald) [inline src/MSBuild/CommandLineSwitches.cs]: likewise, drop this comment.

**Comment by rainersigwald** [comment]
@dotnet-bot test OSX Build for CoreCLR please

**Comment by mfilippov** [comment]
All review issues fixed.

**Pull request merged**